components/helm-broker/internal/controller/addonsconfiguration_controller.go

```diff
@@ -0,0 +1,500 @@
+	docsTopicProvider docsProvider
+
+	protection     protection
+	bundleProvider bundleProvider
+
+	// syncBroker informs ServiceBroker should be resync, it should be true if
+	// operation insert/delete was made on storage
+	syncBroker  bool
+	developMode bool
+}
+
+// NewReconcileAddonsConfiguration returns a new reconcile.Reconciler
+func NewReconcileAddonsConfiguration(mgr manager.Manager, bp bundleProvider, brokerFacade brokerFacade, chartStorage chartStorage, bundleStorage bundleStorage, developMode bool, docsTopicProvider docsProvider, brokerSyncer brokerSyncer) reconcile.Reconciler {
+	return &ReconcileAddonsConfiguration{
+		log:    logrus.WithField("controller", "addons-configuration"),
+		Client: mgr.GetClient(),
+		scheme: mgr.GetScheme(),
+
+		chartStorage:  chartStorage,
+		bundleStorage: bundleStorage,
+
+		bundleProvider: bp,
+		protection:     protection{},
+
+		brokerSyncer:      brokerSyncer,
+		brokerFacade:      brokerFacade,
+		docsTopicProvider: docsTopicProvider,
+
+		developMode: developMode,
+		syncBroker:  false,
+	}
+}
+
+// Reconcile reads that state of the cluster for a AddonsConfiguration object and makes changes based on the state read
+// and what is in the AddonsConfiguration.Spec
+func (r *ReconcileAddonsConfiguration) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	instance := &addonsv1alpha1.AddonsConfiguration{}
+	err := r.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if instance.DeletionTimestamp != nil {
+		if err := r.deleteAddonsProcess(instance); err != nil {
+			return reconcile.Result{RequeueAfter: time.Second * 15}, exerr.Wrapf(err, "while deleting AddonConfiguration %q", request.NamespacedName)
+		}
+		return reconcile.Result{}, nil
+	}
+
+	if instance.Status.ObservedGeneration == 0 {
+		r.log.Infof("Start add AddonsConfiguration %s/%s process", instance.Name, instance.Namespace)
+		pendingInstance, err := r.setPendingStatus(instance)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		updatedInstance, err := r.addFinalizer(pendingInstance)
+		if err != nil {
+			return reconcile.Result{Requeue: true}, exerr.Wrapf(err, "while adding a finalizer to AddonsConfiguration %q", request.NamespacedName)
+		}
+		err = r.addAddonsProcess(updatedInstance)
+		if err != nil {
+			return reconcile.Result{}, exerr.Wrapf(err, "while creating AddonsConfiguration %q", request.NamespacedName)
+		}
+		r.log.Info("Add AddonsConfiguration process completed")
+
+	} else if instance.Generation > instance.Status.ObservedGeneration {
+		r.log.Infof("Start update AddonsConfiguration %s/%s process", instance.Name, instance.Namespace)
+
+		instance.Status = addonsv1alpha1.AddonsConfigurationStatus{}
+		err = r.addAddonsProcess(instance)
+		if err != nil {
+			return reconcile.Result{}, exerr.Wrapf(err, "while updating AddonsConfiguration %q", request.NamespacedName)
+		}
+		r.log.Info("Update AddonsConfiguration process completed")
```

Comment:
What about the rest of the `AddonsConfiguration`. Should they be reprocessed again?

Reply:
that's a good question, let's discuss